packages/enhanced/src/lib/container/runtime/EmbedFederationRuntimeModule.ts

```diff
@@ -0,0 +1,96 @@
+import { normalizeWebpackPath } from '@module-federation/sdk/normalize-webpack-path';
+import { getFederationGlobalScope } from './utils';
+import type { Chunk, Module } from 'webpack';
+
+const { RuntimeModule, NormalModule, Template, RuntimeGlobals } = require(
+  normalizeWebpackPath('webpack'),
+) as typeof import('webpack');
+const ConcatenatedModule = require(
+  normalizeWebpackPath('webpack/lib/optimize/ConcatenatedModule'),
+) as typeof import('webpack/lib/optimize/ConcatenatedModule');
+
+const federationGlobal = getFederationGlobalScope(RuntimeGlobals);
+
+class EmbedFederationRuntimeModule extends RuntimeModule {
+  private bundlerRuntimePath: string;
+
+  constructor(bundlerRuntimePath: string) {
+    super('EmbedFederationRuntimeModule', RuntimeModule.STAGE_ATTACH);
+    this.bundlerRuntimePath = bundlerRuntimePath;
+  }
+
+  override identifier() {
+    return 'webpack/runtime/embed/federation';
+  }
+
+  override generate(): string | null {
+    const { compilation, chunk, chunkGraph, bundlerRuntimePath } = this;
+    if (!chunk || !chunkGraph || !compilation) {
+      return null;
+    }
+
+    const found = this.findModule(chunk, bundlerRuntimePath);
```

Comment:
So you still set `import federation from 'bundlerRuntimePath'` in `.federation/entry` , and find the module id and then embed it to runtimeModule, am i understand right ?

Reply:
Yes - the "hoisted" solution does so by finding the built module in the entry - then moves it to the runtime so it can be eager required. So this embed runtime is still from the actual bundle and currently i still rely on entry to import .federation/entry. 

This can be improved in the future - i did discover some solutions but this is less complex initial approach.
I think we could offer external runtime as well if embedRuntime: external | hoisted | false - so when entry imports runtime, we could have resolve for module so it actually imports module.exports = __FEDERATION__.externalizedRuntime or something